src/main/java/com/amazon/dlic/auth/ldap/util/LdapHelper.java

```diff
@@ -34,37 +34,46 @@
 import org.ldaptive.referral.SearchReferralHandler;
 
 import org.opensearch.SpecialPermission;
+import org.opensearch.common.settings.Settings;
 
 public class LdapHelper {
 
     private static SearchFilter ALL = new SearchFilter("(objectClass=*)");
+
     @SuppressWarnings("removal")
     public static List<LdapEntry> search(final Connection conn, final String unescapedDn, SearchFilter filter,
-            final SearchScope searchScope, final String[] returnAttributes) throws LdapException {
+            final SearchScope searchScope, final String[] returnAttributes, Settings settings) throws LdapException {
```

Comment:
searching an Ldap connection doesn't seem like it should require an OpenSearch.Settings object, instead of passing the settings all the way down, we should pass a parameter to follow referrals or not